### ISSUE TITLE
Can not view text files on Mac Safari w/ private browsing mode

### ISSUE THREAD

**Issue by javabean** [silverwind/droppy#292]
Trying to click a text file when Droppy is configured readOnly does nothing. It should probably display the file content in a "read-only" file viewer. :-)

**Comment by silverwind** [comment]
Hmm, I cannot reproduce this. Does it really work when you switch off read-only mode? Which browser are you using? Any errors visible in the browser console or on the server log?

BTW, I made a few unrelated improvements to readOnly mode in v6.3.4.

**Comment by javabean** [comment]
Issue confirmed with 6.3.4, including without `readOnly` mode! :-)  
But you are right: *this is an issue with Safari browser on a Mac* (Firefox, Chrome, and iOS Safari all work fine)! I have updated this issue title.

Using Safari, when I click on a `.txt` file, I see in Droppy logs:
> 2017-03-26 09:56:15 [INFO] 87.88.10.184:41310 GET /!/type/test.txt [200] [0ms]
> 2017-03-26 09:56:15 [INFO] 87.88.10.184:41310 GET /!/file/test.txt [200] [6ms]
> 2017-03-26 09:56:15 [DEBG] 172.31.11.3:41252 RECV { vId: 0, type: 'REQUEST_UPDATE', data: '/', token: '0c105cf48af657ab9de8f4cf4e3e8a66' }
> 2017-03-26 09:56:15 [DEBG] 172.31.11.3:41252 SEND { type: 'UPDATE_DIRECTORY', vId: 0, folder: '/', data: { <snip snip private data>, 'test.txt': 'f|1391256801|613', <snip snip private data> } }

My Droppy `config.json`:
```
{
  "listeners": [
    {
      "host": [ "0.0.0.0" ],
      "port": 8989,
      "protocol": "http"
    }
  ],
  "public": true,
  "timestamps": true,
  "linkLength": 5,
  "logLevel": 3,
  "maxFileSize": 0,
  "updateInterval": 1000,
  "pollingInterval": 900000,
  "keepAlive": 20000,
  "allowFrame": false,
  "readOnly": false,
  "compression": false
}
```

(TLS and auth are both handled by an Apache httpd proxy running in front of your `silverwind/droppy` Docker container.)

**Comment by silverwind** [comment]
Must be something Apache related I guess. Does https://droppy.silverwind.io/#/example.html work for you? I've configured it pretty similar to you but it is behind nginx.

**Comment by javabean** [comment]
This is a pure Safari bug: I tested with your own instance, as well as a bare Docker container (without any proxy in the middle) to no avail. There are no log in the JavaScript console.  
Workaround for now: use another browser. :-)

Do you have access to a Mac? If not, let me know how I can assist you investigating.

**Comment by silverwind** [comment]
Yes, I have the latest Safari:

<img width="914" src="https://cloud.githubusercontent.com/assets/115237/24334472/0086c494-126b-11e7-87e2-d85f0a008937.png">

What's your version? Maybe some extension interfering?

**Comment by javabean** [comment]
I have Safari 10.0.3 / macOS Sierra 10.12.3. Also tried with Safari Technology Preview 10.2.  
I also disabled all plugins, for the same result.

*But* I finally found out how to reproduce the problem: private browsing!  
Without private browsing mode, all works perfectly well.

**Comment by silverwind** [comment]
Ah, now we're getting something, I can reproduce it too in private browsing mode.

Looks like Safari blocks access to localStorage in that mode which means droppy can't save user preferences and encounters an error when trying to load any of them. I have to investigate if saving preferences is still somehow possible and then handle the errors properly.

**Comment by silverwind** [comment]
Fixed in v6.3.5. Settings won't be saved across sessions in private browsing mode, but I guess that's actually expected in that mode.

**Comment by javabean** [comment]
Woa, that was quick! Thanks @silverwind! 🥇
